cmd/queue/main.go

```diff
@@ -435,9 +435,9 @@ func buildMetricsServer(promStatReporter *queue.PrometheusStatsReporter, protobu
 func metricsHTTPHandler(promStatReporter *queue.PrometheusStatsReporter, protobufStatReporter *queue.ProtobufStatsReporter) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		if strings.Contains(r.Header.Get("Accept"), network.ProtoAcceptContent) {
-			protobufStatReporter.Handler().ServeHTTP(w, r)
+			protobufStatReporter.ServeHTTP(w, r)
```

Comment:
I guess if we wanted to keep the old behavior (explicit handler) we should preallocate those two before the closure to avoid an allocation everytime. I'll leave that to your judgement though.

Reply:
my assumption is once we drop prometheus support we can just directly use ServeHTTP to implement http.Handler and drop this whole thing?